python/cudf/cudf/core/dataframe.py

```diff
@@ -1550,8 +1550,18 @@ def _concat(
 
         # Reassign precision for any decimal cols
         for name, col in out._data.items():
-            if isinstance(col, cudf.core.column.Decimal64Column):
-                col = col._with_type_metadata(tables[0]._data[name].dtype)
+            if isinstance(
+                col,
+                (
+                    cudf.core.column.Decimal64Column,
+                    cudf.core.column.StructColumn,
```

Comment:
This is a common pattern:
https://github.com/rapidsai/cudf/blob/813ac97b2143c8d1d8ca95435863f5234408a681/python/cudf/cudf/core/frame.py#L2154-L2157

To answer why only these two types are being handled separately is because `_reassign_categories` predates `_with_type_metadata` and will likely need a cleanup with some careful benchmarking to eliminate `_reassign_categories` which are out of the scope of this PR.